src-self-hosted/backend/x86.zig

```diff
@@ -0,0 +1,32 @@
+// zig fmt: off
+pub const Register = enum(u8) {
+    // 0 through 7, 32-bit registers. id is int value
+    eax, ecx, edx, ebx, esp, ebp, esi, edi, 
+
+    // 8-15, 16-bit registers. id is int value - 8.
+    ax, cx, dx, bx, sp, bp, si, di,
+    
+    // 16-23, 8-bit registers. id is int value - 16.
+    al, bl, cl, dl, ah, ch, dh, bh,
```

Comment:
is missing `ah, bh, ch, ` etc intentional?

Reply:
They're in the correct order, yes.

Reply:
(For legacy reasons, the order is A C D B, not A B C D. This is due to the inheritance from processors with the AF, BC, DE, and HL register pairs; AF -> A, BC -> C, DE -> D, HL -> B)

Reply:
sorry I was just blind: `*h` registers are right in this line.